spec/models/user_spec.rb

```diff
@@ -475,6 +475,22 @@ def assert_good(username)
       expect(User.username_available?('DonKey')).to eq(false)
       expect(User.username_available?('test')).to eq(false)
     end
+
+    it 'tests when a username is reserved by an expression with wildcard' do
+      SiteSetting.reserved_usernames = 'test)|*admin*|foo*|*bar|abc.def'
```

Comment:
We'll need to update the description for the site settings to clearly indicate that Regexp expressions are allowed.